src/modbus-rtu.c

```diff
@@ -1228,15 +1236,27 @@ modbus_t* modbus_new_rtu(const char *device,
         return NULL;
     }
 
-    ctx = (modbus_t *)malloc(sizeof(modbus_t));
+    if ((ctx = (modbus_t *)malloc(sizeof(modbus_t))) == NULL) {
```

Comment:
below, you use two lines, one to malloc, and one to check, Why not keep the same style here, rather than mixing it all up into one line?